src/core/backbone_router/leader.hpp

```diff
@@ -56,18 +56,28 @@ typedef otBackboneRouterConfig BackboneRouterConfig;
 class Leader : public InstanceLocator
 {
 public:
-    // Primary Backbone Router Service State or State change.
+    // Primary Backbone Router Service state or state change.
     enum State
     {
-        kStateNone = 0,       // Not exist (trigger Backbone Router register its service).
-        kStateAdded,          // Newly added.
-        kStateRemoved,        // Newly removed (trigger Backbone Router register its service).
-        kStateToTriggerRereg, // Short address or sequence number changes (trigger re-registration).
-                              // May also have ReregistrationDelay or MlrTimeout update.
-        kStateRefreshed,      // Only ReregistrationDelay or MlrTimeout changes.
-        kStateUnchanged,      // No change on Primary Backbone Router information (only for logging).
+        kStateNone = 0,       ///< Not exist (trigger Backbone Router register its service).
+        kStateAdded,          ///< Newly added.
+        kStateRemoved,        ///< Newly removed (trigger Backbone Router register its service).
+        kStateToTriggerRereg, ///< Short address or sequence number changes (trigger re-registration).
+                              ///< May also have ReregistrationDelay or MlrTimeout update.
+        kStateRefreshed,      ///< Only ReregistrationDelay or MlrTimeout changes.
+        kStateUnchanged,      ///< No change on Primary Backbone Router information (only for logging).
     };
 
+    // Domain Prefix state or state change.
+    typedef enum DomainPrefixState
```

Comment:
```suggestion
   enum DomainPrefixState
```
I think no need for `typedef` here. `enum DomainPrefixState` would already define the type (since this is C++ only header).